code/modules/events/tram_malfunction.dm

```diff
@@ -42,8 +42,8 @@
 
 		signal.obj_flags |= EMAGGED
 
-		if(signal.signal_state != XING_STATE_GREEN)
-			signal.set_signal_state(XING_STATE_GREEN)
+		if(signal.signal_state != XING_STATE_MALF)
```

Comment:
This should be in a shared defines file instead of copying and pasting between the two

But more likely this should just be a proc like `signal.set_malfunctioning()`

Reply:
Done